ddtrace/contrib/kafka/patch.py

```diff
@@ -184,7 +205,15 @@ def traced_poll(func, instance, args, kwargs):
             message_key = message.key() or ""
             message_offset = message.offset() or -1
             span.set_tag_str(kafkax.TOPIC, message.topic())
-            span.set_tag_str(kafkax.MESSAGE_KEY, ensure_text(message_key, errors="replace"))
+
+            # If this is a deserializing consumer, do not set the key as a tag since we
+            # do not have the serialization function
+            if (
+                not isinstance(instance, _DeserializingConsumer)
+                or isinstance(message_key, str)
+                or isinstance(message_key, bytes)
+            ):
+                span.set_tag_str(kafkax.MESSAGE_KEY, message_key)
```

Comment:
What are the drawbacks of not setting this tag for `_DeserializingConsumer`? Can this break DSM? Do we need to call out this gap in our support in our public docs?

Reply:
AFAIK we do not use the `message.key` tag for anything other than more information tagged on the span.

Reply:
And at this moment if the key is not a string then we are already not setting it